### PR TITLE
Remove sync from UI thread

### PR DESCRIPTION
**Pull request by MatkovIvan**
* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you ~add~ remove unit tests?
* [ ] ~Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?~
* [x] Did you check UI tests on the sample app? They are not executed on CI.

## Description

During the testing I found that the line with `sendLogs` call can take up to 300-400ms (!), that unacceptable for UI thread. So, I've marked this issue as a bug.
This happens because of call synchronized method (`sendLogs` and status check).

## Related PRs or issues

#374, #1421
[AB#81136](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/81136)

### CODE REVIEW

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/channel/DefaultChannel.java [hunk 1]
```diff
@@ -132,7 +134,7 @@
      * State checker. If this counter changes during an async call, we have to ignore the result in the callback.
      * Cancelling a database call would be unreliable, and if it's too fast you could still have the callback being called.
      */
-    private int mCurrentState;
```
Comment (guperrot) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/channel/DefaultChannel.java:135]: This state check code was created before the introduction of the background looper. How many methods from this class are not called from the background looper?

File: sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java [hunk 2]
```diff
@@ -1128,20 +1128,12 @@ public void onBeforeCalling(URL url, Map<String, String> headers) {
...
-                    public void run() {
-                        try {
-                            String payload = httpResponse.getPayload();
-                            handleApiCallSuccess(releaseCallId, payload, ReleaseDetails.parse(payload), distributionGroupId);
-                        } catch (JSONException e) {
-                            onCallFailed(e);
-                        }
-                    }
-                });
+                try {
```
Comment (guperrot) [inline sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java:1131]: When was the android bug fixed?

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/channel/DefaultChannel.java [hunk 3]
```diff
@@ -493,27 +494,7 @@ private synchronized void triggerIngestion(final @NonNull GroupState groupState)
 
         /* Remember this batch. */
         groupState.mSendingBatches.put(batchId, batch);
-
-        /*
-         * Due to bug on old Android versions (verified on 4.0.4),
```
Comment (guperrot) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/channel/DefaultChannel.java:498]: Was the async bug tested on 4.1? The link to the android source code is not clear about released versions, just wanting to make sure.

Comment (MatkovIvan) [reply sdk/appcenter/src/main/java/com/microsoft/appcenter/channel/DefaultChannel.java:498]: Yes (on emulator). Also, that commit (android source) contains comment about the versions:

> The AsyncTask class must be loaded on the UI thread. This is done
> automatically as of {@link android.os.Build.VERSION_CODES#JELLY_BEAN}.

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/channel/DefaultChannel.java [hunk 4]
```diff
@@ -525,46 +505,44 @@ public void run() {
      * @param batchId      The batch ID.
      */
     @MainThread
-    private synchronized void sendLogs(final GroupState groupState, final int currentState, List<Log> batch, final String batchId) {
-        if (checkStateDidNotChange(groupState, currentState)) {
+    private void sendLogs(final GroupState groupState, final int currentState, List<Log> batch, final String batchId) {
```
Comment (guperrot) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/channel/DefaultChannel.java:508]: is currentState parameter unused?, should cleanup if yes.

Comment (MatkovIvan) [reply sdk/appcenter/src/main/java/com/microsoft/appcenter/channel/DefaultChannel.java:508]: No, it's used for calling `checkPendingLogsAfterPost`

**Pull request merged**